examples/pytorch/pytorch-cifar10-sagemaker/source/cifar10.py

```diff
@@ -49,112 +14,152 @@ class Net(nn.Module):
-
-    # Let us look at how the network performs on the whole dataset.
-
-    correct = 0
-    total = 0
-    with torch.no_grad():
-        for data in testloader:
-            images, labels = data
-            images, labels = images.to(device), labels.to(device)
-
-            outputs = net(images)
-            _, predicted = torch.max(outputs.data, 1)
-            total += labels.size(0)
-            correct += (predicted == labels).sum().item()
-
-    test_acc = 100 * correct / total
-
-    class_correct = list(0. for i in range(10))
-    class_total = list(0. for i in range(10))
-    with torch.no_grad():
-        for data in testloader:
-            images, labels = data
-            images, labels = images.to(device), labels.to(device)
-
-            outputs = net(images)
-            _, predicted = torch.max(outputs, 1)
-            c = (predicted == labels).squeeze()
-            for i in range(4):
-                label = labels[i]
-                class_correct[label] += c[i].item()
-                class_total[label] += 1
-
-    print("Test Accuracy: %.4f" % test_acc)
-    class_acc = {}
-    for i in range(10):
-        print('Accuracy of %5s : %2d %%' % (
-            classes[i], 100 * class_correct[i] / class_total[i]))
-        class_acc["Accuracy of %5s" %
-                  (classes[i])] = 100 * class_correct[i] / class_total[i]
-
-    wandb.log(class_acc, commit=False)
-    wandb.log({"Examples": example_images,
+def main():
+    wandb.init(project="sm-pytorch-cifar")
+
+    config = wandb.config
+    # Set defaults if we dont have values from SageMaker
+    if config.get('batch_size') is None:
+        config.batch_size = 20
+        config.lr = 0.001
+        config.momentum = 0.9
+        config.epochs = 10
+        config.hidden_nodes = 120
+        config.conv1_channels = 5
+        config.conv2_channels = 16
+
+
+    ########################################################################
+    # The output of torchvision datasets are PILImage images of range [0, 1].
+    # We transform them to Tensors of normalized range [-1, 1].
+
+    transform = transforms.Compose(
+        [transforms.ToTensor(),
+        transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+
+    trainset = torchvision.datasets.CIFAR10(
+        os.getenv('SM_CHANNEL_TRAINING', 'data'), train=True, transform=transform)
+    trainloader = torch.utils.data.DataLoader(trainset, batch_size=config.batch_size,
+                                          shuffle=True, num_workers=2)
+
+    testset = torchvision.datasets.CIFAR10(
+        os.getenv('SM_CHANNEL_TRAINING', 'data'), train=False, transform=transform)
+    testloader = torch.utils.data.DataLoader(testset, batch_size=config.batch_size,
+                                         shuffle=False, num_workers=2)
+
+    classes = ('plane', 'car', 'bird', 'cat',
+            'deer', 'dog', 'frog', 'horse', 'ship', 'truck')
+
+    net = Net()
```

Comment:
should we pass the config to `Net` instead of having the `wandb.config` object?
what is the way we prefer our user to use this object?